File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        Map<Character, Integer> keyCount = new HashMap<>();
        
        for (String key : keymap) {
            for (int i = 0; i < key.length(); i++) {
                char c = key.charAt(i);
                if (!keyCount.containsKey(c) || i < keyCount.get(c)) {
                    keyCount.put(c, i + 1);
                }
            }
        }
        
        int len = targets.length;
        int[] answer = new int[len];
        
        for (int i = 0; i < len; i++) {
            int subtotal = 0;
            boolean canType = true;
            
            for (char c : targets[i].toCharArray()) {
                if (!keyCount.containsKey(c)) {
                    canType = false;
                    break;
                }
                
                subtotal += keyCount.get(c);
            }
            
            answer[i] = canType ? subtotal : -1;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.kt
class Solution {
    fun solution(keymap: Array<String>, targets: Array<String>): IntArray {
        val MAX_LEN = 100
        val keyCount = mutableMapOf<Char, Int>()
        
        for (key in keymap) {
            key.forEachIndexed { idx, c ->
                keyCount[c] = minOf(keyCount.getOrDefault(c, MAX_LEN), idx + 1)
            }
        }
        
        return targets.map { target ->
            var subtotal = 0
            for (c in target) {
                val press = keyCount[c]
                if (press == null) {
                    return@map -1
                }
                subtotal += press
            }
            subtotal
        }.toIntArray()
    }
}